internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybrid.java

```diff
@@ -2081,6 +2082,68 @@ public void testHybridWithPartitionWiseConsumer() throws Exception {
     }
   }
 
+  @Test(timeOut = 180 * Time.MS_PER_SECOND)
+  public void testLeaderShouldCalculateRewindDuringPromotion() {
```

Comment:
Thanks for adding an integ test. Just out of curiosity, how reliable have you seen this test be at catching the issue in the absence of the main code fix?

Reply:
I copy the two tests I added to main, they all breaks as it re-consumes from beginning of the RT topic and take in bad key.